core/Symbols.h

```diff
@@ -602,10 +602,20 @@ class ClassOrModule final {
     ClassOrModule deepCopy(const GlobalState &to, bool keepGsId = false) const;
     void sanityCheck(const GlobalState &gs) const;
 
+    inline bool isBehaviorDefining() {
+        return isBehaviorDefining_;
+    }
+
+    inline void markAsBehaviorDefining() {
+        isBehaviorDefining_ = true;
+    }
+
 private:
     friend class serialize::SerializerImpl;
     friend class GlobalState;
 
+    bool isBehaviorDefining_ = false;
```

Comment:
Original plan was to have it as a bit inside flags, but then in order to not serialize it, it felt like I'd have to make changes where I'm mindful of endian-ness. Could be wrong.